### PR TITLE
Add support for const attribute in libgccjit

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: gcc/jit/jit-playback.cc [hunk 1]
```diff
@@ -675,6 +675,9 @@ new_function (location *loc,
     /* See handle_pure_attribute in gcc/c-family/c-attribs.cc. */
     else if (attr == GCC_JIT_FN_ATTRIBUTE_PURE)
       DECL_PURE_P (fndecl) = 1;
+    /* See handle_pure_attribute in gcc/c-family/c-attribs.cc. */
```
Comment (antoyo) [inline gcc/jit/jit-playback.cc:678]: Type: you wrote `pure` instead of whatever the function is called.

**Review by antoyo — COMMENTED** [review]
I guess you didn't add the case in `fn_attribute_to_string` because const is not really an attribute?

File: gcc/jit/jit-playback.cc (hunk 1 above)
Comment (GuillaumeGomez) [reply gcc/jit/jit-playback.cc:678]: Woups.

File: gcc/testsuite/jit.dg/test-const-attribute.c [hunk 2]
```diff
@@ -0,0 +1,134 @@
+/* { dg-do compile { target x86_64-*-* } } */
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "libgccjit.h"
+
+/* We don't want set_options() in harness.h to set -O3 to see that the cold
```
Comment (antoyo) [inline gcc/testsuite/jit.dg/test-const-attribute.c:8]:
```suggestion
/* We don't want set_options() in harness.h to set -O3 to see that the const
```

**Review by antoyo — APPROVED** [review]

**Pull request merged**